aptos-move/framework/supra-framework/sources/multisig_voting.spec.move

```diff
@@ -0,0 +1,427 @@
+        aborts_if has_multi_step_key && from_bcs::deserialize<bool>(simple_map::spec_get(proposal.metadata, multi_step_key));
+
+        let post post_voting_forum = global<VotingForum<ProposalType>>(voting_forum_address);
+        let post post_proposal = table::spec_get(post_voting_forum.proposals, proposal_id);
+        aborts_if !exists<timestamp::CurrentTimeMicroseconds>(@supra_framework);
+        // property 3: Ensure that proposal is successfully resolved.
+        /// [high-level-req-3]
+        ensures post_proposal.is_resolved == true;
+        ensures post_proposal.resolution_time_secs == timestamp::spec_now_seconds();
+
+        aborts_if option::spec_is_none(proposal.execution_content);
+        ensures result == option::spec_borrow(proposal.execution_content);
+        ensures option::spec_is_none(post_proposal.execution_content);
+    }
+
+    spec resolve_proposal_v2 {
+        use supra_framework::chain_status;
+        pragma verify_duration_estimate = 300;
+        // Ensures existence of Timestamp
+        requires chain_status::is_operating();
+
+        include IsProposalResolvableAbortsIf<ProposalType>;
+        let voting_forum = global<VotingForum<ProposalType>>(voting_forum_address);
+        let proposal = table::spec_get(voting_forum.proposals, proposal_id);
+        let post post_voting_forum = global<VotingForum<ProposalType>>(voting_forum_address);
+        let post post_proposal = table::spec_get(post_voting_forum.proposals, proposal_id);
+        let multi_step_in_execution_key = std::string::spec_utf8(IS_MULTI_STEP_PROPOSAL_IN_EXECUTION_KEY);
+        aborts_if !std::string::spec_internal_check_utf8(IS_MULTI_STEP_PROPOSAL_IN_EXECUTION_KEY);
+        aborts_if !std::string::spec_internal_check_utf8(IS_MULTI_STEP_PROPOSAL_KEY);
+        ensures (simple_map::spec_contains_key(proposal.metadata, multi_step_in_execution_key) && len(next_execution_hash) != 0) ==>
+            simple_map::spec_get(post_proposal.metadata, multi_step_in_execution_key) == std::bcs::serialize(true);
+        ensures (simple_map::spec_contains_key(proposal.metadata, multi_step_in_execution_key) &&
+            (len(next_execution_hash) == 0 && !is_multi_step)) ==>
+            simple_map::spec_get(post_proposal.metadata, multi_step_in_execution_key) == std::bcs::serialize(true);
+
+        let multi_step_key = std::string::spec_utf8(IS_MULTI_STEP_PROPOSAL_KEY);
+        aborts_if simple_map::spec_contains_key(proposal.metadata, multi_step_key) &&
+            !from_bcs::deserializable<bool>(simple_map::spec_get(proposal.metadata, multi_step_key));
+        let is_multi_step = simple_map::spec_contains_key(proposal.metadata, multi_step_key) &&
+            from_bcs::deserialize(simple_map::spec_get(proposal.metadata, multi_step_key));
+        aborts_if !is_multi_step && len(next_execution_hash) != 0;
+
+        aborts_if len(next_execution_hash) == 0 && !exists<timestamp::CurrentTimeMicroseconds>(@supra_framework);
```

Comment:
Again, not sure at this point why `!exist<timestamp::CurrentTimeMicroseconds>(@supra_framework)` is needed. I do not see this in the source code.

Reply:
This is for checking the whether time already started. This is required if we have call to timestamp::now_seconds()